internal/service/resourcepolicy/resource.go

```diff
@@ -35,6 +38,40 @@ type resourcePolicyRS struct {
 	config.RSCommon
 }
 
+var charSetAlphaNum = []rune("abcdefghijklmnopqrstuvwxyz012346789")
+
+// randPolicyName to workaround for POLICY_CANNOT_CONTAIN_A_DUPLICATE_NAME error until https://jira.mongodb.org/browse/CLOUDP-272944 is resolved.
+func randPolicyName(n int) string {
+	b := make([]rune, n)
+	for i := range b {
+		b[i] = charSetAlphaNum[rand.Intn(len(charSetAlphaNum))] //nolint:gosec // This is not a security-sensitive operation, only to avoid POLICY_CANNOT_CONTAIN_A_DUPLICATE_NAME error
+	}
+	return string(b)
+}
+
+func (r *resourcePolicyRS) ModifyPlan(ctx context.Context, req resource.ModifyPlanRequest, resp *resource.ModifyPlanResponse) {
+	var policies []TFPolicyModel
+	resp.Diagnostics.Append(req.Plan.GetAttribute(ctx, path.Root("policies"), &policies)...)
+	sdkPolicies := NewAdminPolicies(ctx, policies)
+	var orgID *string
+	resp.Diagnostics.Append(req.Plan.GetAttribute(ctx, path.Root("org_id"), &orgID)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	if orgID == nil {
+		return
+	}
+	sdkCreate := &admin.ApiAtlasResourcePolicyCreate{
+		Name:     randPolicyName(16),
```

Comment:
isn't the name something that gets passed from the customer? I am probably missing some context here.

Reply:
Let me know if this issue https://jira.mongodb.org/browse/CLOUDP-274990 makes it clear